File: frontend/lib/features/attendance/models/attendance_mark.dart
/// 하루치 출석 표시 상태.
///
/// 백엔드 constant.AttendanceMark(PRESENT/ABSENT/VACATION/UPCOMING)와
/// 1:1로 매핑된다(openapi.yaml AttendanceMark 스키마, docs/API_CONTRACT.md
/// "5. 스터디 출석" 참고). AttendanceApiDataSourceImpl이 서버 응답을 파싱할
/// 때 [fromJson]을, 필요 시 [toJson]을 쓴다.
enum AttendanceMark {
  /// 출석.
  present,

  /// 결석.
  absent,

  /// 휴가 처리됨(결석으로 집계하지 않음).
  vacation,

  /// 아직 오지 않은 날(이번 주의 미래 요일) 또는 아직 체크되지 않음.
  upcoming;

  bool get isPast => this == present || this == absent || this == vacation;

  static AttendanceMark fromJson(String value) {
    switch (value) {
      case 'PRESENT':
        return AttendanceMark.present;
      case 'ABSENT':
        return AttendanceMark.absent;
      case 'VACATION':
        return AttendanceMark.vacation;
      case 'UPCOMING':
        return AttendanceMark.upcoming;
      default:
        // 서버가 아직 모르는 새 값을 내려주더라도 화면이 죽지 않도록, 안전한
        // 쪽(아직 정해지지 않음)으로 방어한다.
        return AttendanceMark.upcoming;
    }
  }

  String toJson() => name.toUpperCase();
}

File: backend/src/main/java/com/moa/constant/AttendanceMark.java
package com.moa.constant;

/**
 * 하루치 출석을 화면에 표시할 때 쓰는 상태. AttendanceOverviewResponse/
 * MyStudyInfoResponse가 이 값을 내려준다.
 *
 * AttendanceStatus(실제 저장되는 값: PRESENT/VACATION)와 달리 ABSENT/UPCOMING을
 * 포함한다 — 이 둘은 attendance_records에 저장되지 않고 AttendanceService가
 * "그 날짜에 행이 있는지 + 오늘/미래인지"를 보고 계산해서 채운다.
 */
public enum AttendanceMark {
    /** 출석. */
    PRESENT,

    /** 결석(그 날짜에 attendance_records 행이 없음, 지나간 날짜). */
    ABSENT,

    /** 휴가 처리됨(결석으로 집계하지 않음). */
    VACATION,

    /** 아직 오지 않은 날, 또는 오늘인데 아직 출석/휴가를 정하지 않음. */
    UPCOMING
}
